src/shared/watchdog.c

```diff
@@ -173,6 +150,11 @@ int watchdog_setup(usec_t timeout) {
         if (watchdog_fd >= 0 && watchdog_timeout == timeout)
                 return 0;
 
+        if (timeout == 0) {
+                watchdog_close(true);
+                return 0;
+        }
```

Comment:
this block should be before the first if check, no?

Reply:
I don't think it makes any difference in practice but it might be easier to read. I'll change it.